### PR TITLE
Change ECM and Hacker Tanks

### PR DESCRIPTION
**Pull request by WilliamWeberBerrutti**
This PR changes the following:

- changes prerequisite for both tanks to be Factory, Radar and Trader
- balances ECM Tank by increasing disabling duration, having seeker projectile, allowing disabling only for enemy units, increasing HP, increasing speed and turnspeed and changing armor type
- balances Hacker Tank by removing damage from its projectile, changing armor type, increasing HP and cost and reducing speed
- increases LOS of Radar Tank and Saucer
- prevents empty resource tankers from being hacked and disabled
- makes Radar Tank selectable with military units

### CODE REVIEW

File: mods/hv/rules/vehicles.yaml [hunk 1]
```diff
@@ -579,7 +579,7 @@ ECMTANK:
 		Class: ecmtank
 	Buildable:
 		Queue: Tank
-		Prerequisites: ~factory.sc, techcenter, trader
+		Prerequisites: ~factory.yi, radar, trader
```
Comment (Mailaender) [inline mods/hv/rules/vehicles.yaml:582]:
```suggestion
		Prerequisites: ~factory.sc, radar, trader
```

Comment (OcelotWalrus) [reply mods/hv/rules/vehicles.yaml:582]: ECM tank is for yuruki right?

Comment (OcelotWalrus) [reply mods/hv/rules/vehicles.yaml:582]: yes he's right. Countermeasure Tank is for synapol, as Hacker Tank is for yuruki.

File: mods/hv/weapons/energetic.yaml [hunk 2]
```diff
@@ -469,9 +469,6 @@ HackBack:
 		DistortionAnimation: 100
 		SegmentLength: 0c256
 		ZOffset: 2048
-	Warhead@Damage: SpreadDamage
```
Comment (GraionDilach) [inline mods/hv/weapons/energetic.yaml:472]: Are you certain this still works?

I haven't checked it ingame yet, but years ago OpenRA couldn't fire weapons at all if they lacked a warhead.

Honestly, I'd just keep this as

```
	Warhead@Damage: SpreadDamage
		ValidTargets: Vulnerability
		Damage: 1
		Spread: 1
		Falloff: 0, 0
```

Comment (WilliamWeberBerrutti) [reply mods/hv/weapons/energetic.yaml:472]: Agreed! Better safe than sorry!

Comment (WilliamWeberBerrutti) [reply mods/hv/weapons/energetic.yaml:472]: What if `Damage: 0` was used?

File: mods/hv/languages/rules/en.ftl [hunk 3]
```diff
@@ -726,7 +726,7 @@ actor-ecmtank =
    .name = Countermeasure Tank
    .generic-name = ECM Tank
    .description = Disables units for a brief moment.
-      Jams incoming missiles.
+      Disrupts satellite video links\nand obscures the battlefield.
```
Comment (OcelotWalrus) [inline mods/hv/languages/rules/en.ftl:729]:
```suggestion
      Disrupts satellite video links
      and obscures the battlefield.
```

File: mods/hv/languages/rules/en.ftl [hunk 4]
```diff
@@ -756,8 +756,7 @@ actor-missiletank =
 
 actor-hackertank =
    .description = Temporarily changes allegiance of targeted units.
-      Disrupts satellite video links
-      and obscures the battlefield when deployed.
+      Jams incoming missiles.
```
Comment (OcelotWalrus) [inline mods/hv/languages/rules/en.ftl:759]:
```suggestion
      Deflects incoming missiles.
```

Maybe deflect could be better

**Review by OcelotWalrus — CHANGES_REQUESTED** [review]

File: mods/hv/weapons/energetic.yaml (hunk 2 above)
Comment (GraionDilach) [reply mods/hv/weapons/energetic.yaml:472]: While I always used `Damage: 1` with other tricks to round it down to 0 during calculations, I can't find anything in the code suggesting me `Damage: 0` wouldn't work directly.

The goal is to have a warhead which satisfies the `Warhead.IsValidAgainst()` call and both options do that.

Comment (GraionDilach) [reply mods/hv/weapons/energetic.yaml:472]: Oh, I see this is resolved now, I just can't mark the discussion as such.

File: mods/hv/languages/rules/en.ftl [hunk 5]
```diff
@@ -726,7 +726,7 @@ actor-ecmtank =
    .name = Countermeasure Tank
    .generic-name = ECM Tank
    .description = Disables units for a brief moment.
-      Jams incoming missiles.
+      Disrupts satellite video links and obscures the battlefield.
```
Comment (OcelotWalrus) [inline mods/hv/languages/rules/en.ftl:729]:
```suggestion
      Disrupts satellite video links
      and obscures the battlefield.
```

Comment (WilliamWeberBerrutti) [reply mods/hv/languages/rules/en.ftl:729]: Does it really need a line break?

Comment (OcelotWalrus) [reply mods/hv/languages/rules/en.ftl:729]: yeah it does. It'll look too long.

File: mods/hv/languages/rules/en.ftl [hunk 6]
```diff
@@ -1019,8 +1020,7 @@ actor-missiletank =
 
 actor-hackertank =
    .description = Temporarily changes allegiance of targeted units.
-      Disrupts satellite video links
-      and obscures the battlefield when deployed.
+      Deflects incoming missiles.
    .name = Hacker Tank
    .encyclopedia = The hacker tank contains equipment to penetrate firewalls of individual units and interfer with their command and control software so they will accept orders from the enemy and engage in friendly fire. The effective change of allegiance is permanent, until the hacker tank gets destroyed or changes the target. It can only have one controlled unit at a time. Its other ability is to obscure the opponent's battlefield on its surroundings when deployed.
```
Comment (Mailaender) [inline mods/hv/languages/rules/en.ftl:1025]: It should hack the video uplink and obscure the battlefield as stated here.

Comment (WilliamWeberBerrutti) [reply mods/hv/languages/rules/en.ftl]: I'll have to playtest a bit more before resolving this comment.

**Comment by Rampoina** [comment]
@WilliamWeberBerrutti says it shouldn't be merged yet, he wants to do more tests and some changes.

**Comment by WilliamWeberBerrutti** [comment]
> @WilliamWeberBerrutti says it shouldn't be merged yet, he wants to do more tests and some changes.

I think it's good now!

**Pull request merged**